### DISCUSSION TITLE
Unstacking window in yabai

### DISCUSSION THREAD

**Discussion by jasonpott** [asmvik/yabai#783] [Q&A] [answered]
Thank you for all your work on Yabai I think it's awesome.  I have been trying to figure out how to unstack a window in yabai but have not found this explicitly described in the documentation.

I use yabai with skhd and I have been trying to add a shortcut to unstack an app that has been previously stacked. From the documentation it is suggested that a previously stacked window can be unstacked by using '--warp'. 

`
hyper - 0x02: yabai -m window --warp east
`   
Allocating this to a shortcut in my skhd config does not result in the window being unstacked. It does however warp an unstacked window. I have also tried using "first" and "last" in place of "east" but have not been successful.

Many thanks for any advice.

**Comment by rockyzhang24** [comment, marked as answer]
I can confirm that `warp` works. I attached a GIF below. The upper windows are two stacked windows (one terminal window, and one Chrome window). The lower window is the normal one. 
* Firstly I press `alt - >` which binds to `yabai -m window --focus stack.next` to switch between these two windows to show these two stacked windows. 
* Next, I focus on one of the stacked windows (the Chrome window), and then I press `alt + shift - s` which binds to `yabai -m window --warp south`. 

You can see that this Chrome window is unstacked and put next to the normal window.

![unstack](https://user-images.githubusercontent.com/11582667/103484362-20c93f80-4da3-11eb-9cb3-e14de18d57c1.gif)

**Reply by asmvik** [reply]
Additionally, if you do not have any other windows open that you can warp into, just toggle float (`yabai -m window --toggle float`) twice for the window that you want to unstack.

**Reply by szfkamil** [reply]
Maybe `yabai -m window --unstack` could be run as `yabai -m window --toggle float && yabai -m window --toggle float` in the background? It took me some time to figure out how to unstack the window when there's no window to warp into.

I recall you mentioning that you prefer not to abstract users from the actual workings of yabai, so perhaps it would be helpful to include a reference to the wiki, as this information is currently buried here.

**Comment by szfkamil** [comment]
I used the following as a solution:
```
# Unstack focused window if no window to warp into

shift + ctrl - u :  yabai -m window --toggle float && yabai -m window --toggle float
```
